src/types.ts

```diff
@@ -29,8 +29,10 @@ export type DefuFn = <Source extends Input, Defaults extends Input>(
   ...defaults: Defaults[]
 ) => MergeObjects<Source, Defaults>;
 
+type HandledNonObject = boolean | number | null | any[] | undefined
```

Comment:
Maybe we can create an `InputSource` type combining these  `Source`? (also I would export the type then)

Reply:
I like new name. If refactoing input args Source type breaks inference, let's keep it as is.